packages/block-library/src/paragraph/use-enter.js

```diff
@@ -56,22 +57,45 @@ export function useOnEnter( props ) {
 			}
 
 			const order = getBlockOrder( wrapperClientId );
-
-			event.preventDefault();
-
 			const position = order.indexOf( clientId );
 
 			// If it is the last block, exit.
 			if ( position === order.length - 1 ) {
-				moveBlocksToPosition(
-					[ clientId ],
-					wrapperClientId,
-					getBlockRootClientId( wrapperClientId ),
-					getBlockIndex( wrapperClientId ) + 1
-				);
+				let newWrapperClientId = wrapperClientId;
+
+				while (
+					! canInsertBlockType(
+						getBlockName( clientId ),
+						getBlockRootClientId( newWrapperClientId )
+					)
+				) {
+					newWrapperClientId =
+						getBlockRootClientId( newWrapperClientId );
+				}
+
+				if ( typeof newWrapperClientId === 'string' ) {
+					event.preventDefault();
+					moveBlocksToPosition(
+						[ clientId ],
+						wrapperClientId,
+						getBlockRootClientId( newWrapperClientId ),
+						getBlockIndex( newWrapperClientId ) + 1
+					);
+				}
 				return;
 			}
 
+			if (
+				! canInsertBlockType(
+					'core/paragraph',
```

Comment:
I think we can use `getDefaultBlockName()` here and avoid hardcoding the block name.

Reply:
Yeah, must have copied the hardcoded name from below. I'll change it there as well.